mkosi/util.py

```diff
@@ -319,3 +320,15 @@ def try_or(fn: Callable[..., T], exception: type[Exception], default: T) -> T:
         return fn()
     except exception:
         return default
+
+
+def groupby(seq: Sequence[T], key: Callable[[T], S]) -> Sequence[tuple[S, list[T]]]:
+    grouped: dict[S, list[T]] = {}
+
+    for i in seq:
+        if key(i) not in grouped:
+            grouped[key(i)] = []
+
+        grouped[key(i)].append(i)
```

Comment:
Let's only call the key function once per element.
```suggestion
        k = key(i)
        if k not in grouped:
            grouped[k] = []

        grouped[k].append(i)
```